app/celer-sim/CMakeLists.txt

```
@@ -40,88 +40,103 @@ if(NOT BUILD_TESTING)
-if(CELERITAS_USE_ROOT)
-  set(_geant_exporter_env
-    "CELER_EXPORT_GEANT_EXE=$<TARGET_FILE:celer-export-geant>")
-endif()
-set(_env
-  "CELERITAS_DEMO_EXE=$<TARGET_FILE:celer-sim>"
-  "${_geant_exporter_env}"
-  "CELER_DISABLE_PARALLEL=1"
-  "CELER_CORE_GEO=${CELERITAS_CORE_GEO}"
-)
-
-if(CELERITAS_USE_Geant4 AND CELERITAS_USE_HepMC3 AND CELERITAS_USE_Python
-    AND (NOT CELERITAS_CORE_GEO STREQUAL "Geant4")
-    AND (VecGeom_GDML_FOUND OR NOT (CELERITAS_CORE_GEO STREQUAL "VecGeom")))
-  set(_needs_deps)
-else()
-  set(_needs_deps DISABLED true)
-endif()
-if(CELER_DISABLE_DEVICE)
-  set(_needs_device DISABLED true)
-else()
-  set(_needs_device)
-endif()
+#-----------------------------------------------------------------------------#
+function(celer_sim_test test_ext gdml_inp_name hepmc3_inp_name mctruth_name)
+  set(_driver "${CMAKE_CURRENT_SOURCE_DIR}/simple-driver.py")
+  set(_gdml_inp "${CELER_APP_DATA_DIR}/${gdml_inp_name}")
+  set(_hepmc3_inp "${CELER_APP_DATA_DIR}/${hepmc3_inp_name}")
+  set(_mctruth_out "")
+  set(_geant_exporter_env "CELER_EXPORT_GEANT_EXE=")
+  if(CELERITAS_USE_ROOT)
+    set(_geant_exporter_env
+      "CELER_EXPORT_GEANT_EXE=$<TARGET_FILE:celer-export-geant>")
+  endif()
+  set(_env
+    "CELERITAS_DEMO_EXE=$<TARGET_FILE:celer-sim>"
+    "${_geant_exporter_env}"
+    "CELER_DISABLE_PARALLEL=1"
+    "CELER_CORE_GEO=${CELERITAS_CORE_GEO}"
+  )
+  
+  if(CELERITAS_USE_Geant4 AND CELERITAS_USE_HepMC3 AND CELERITAS_USE_Python
+      AND (NOT CELERITAS_CORE_GEO STREQUAL "Geant4")
+      AND (VecGeom_GDML_FOUND OR NOT (CELERITAS_CORE_GEO STREQUAL "VecGeom")))
+    set(_needs_deps)
+  else()
+    set(_needs_deps DISABLED true)
+  endif()
+  if(CELER_DISABLE_DEVICE)
+    set(_needs_device DISABLED true)
+  else()
+    set(_needs_device)
+  endif()
+  
+  if(CELERITAS_REAL_TYPE STREQUAL "float")
+    # FIXME: propagation fails in single precision
+    set(_needs_deps DISABLED true)
+  endif()
+  
+  add_test(NAME "app/celer-sim-${test_ext}:gpu"
+    COMMAND "${CELER_PYTHON}"
+    "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${_mctruth_out}"
+  )
+  set_tests_properties("app/celer-sim-${test_ext}:gpu" PROPERTIES
+    ENVIRONMENT "${_env};${CELER_G4ENV}"
+    RESOURCE_LOCK gpu
+    REQUIRED_FILES "${_driver};${_gdml_inp};${_hepmc3_inp}"
+    LABELS "app;nomemcheck;gpu"
+    ${_needs_deps}
+    ${_needs_device}
+    ${CELER_NEEDS_PYTHON}
+  )
+  
+  # Enable ROOT output for the celer-sim-cpu test when ROOT is available
+  if(CELERITAS_USE_ROOT)
+    set(_mctruth_out "${CMAKE_CURRENT_BINARY_DIR}/${mctruth_name}")
+  endif()
+  set(CELER_GEO ${CELERITAS_CORE_GEO})
+  if(CELERITAS_CORE_GEO STREQUAL "ORANGE" AND NOT CELERITAS_USE_Geant4)
+    # Signal to the app that we have to use the JSON input
+    set(CELER_GEO "${CELER_CORE_GEO}-JSON")
+  endif()
+  
+  add_test(NAME "app/celer-sim-${test_ext}:cpu"
+    COMMAND "${CELER_PYTHON}"
+    "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${mctruth_name}"
```

I think this should be:
```suggestion
    "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${mctruth_out}"
```